sched/semaphore/sem_wait.c

```diff
@@ -251,37 +251,21 @@ int nxsem_wait_uninterruptible(FAR sem_t *sem)
 
 int sem_wait(FAR sem_t *sem)
 {
-  int errcode;
   int ret;
 
   /* sem_wait() is a cancellation point */
 
-  if (enter_cancellation_point())
-    {
-#ifdef CONFIG_CANCELLATION_POINTS
-      /* If there is a pending cancellation, then do not perform
-       * the wait.  Exit now with ECANCELED.
-       */
-
-      errcode = ECANCELED;
-      goto errout_with_cancelpt;
-#endif
-    }
```

Comment:
My guess is that this same pattern exists in all cancellation point functions after enter_cancellation_point() is called.  Shouldn't they all be changed?

Reply:
For example this is clock wait:
```
int sem_clockwait(FAR sem_t *sem, clockid_t clockid,
                  FAR const struct timespec *abstime)
{
  int ret;

  /* sem_timedwait() is a cancellation point */

  enter_cancellation_point();

  /* Let nxsem_timedout() do the work */

  ret = nxsem_clockwait(sem, clockid, abstime);
  if (ret < 0)
    {
      set_errno(-ret);
      ret = ERROR;
    }

  leave_cancellation_point();
  return ret;
}
```

Reply:
Yeah, sorry not fully true:
```
static int uart_tcdrain(FAR uart_dev_t *dev,
                        bool cancelable, clock_t timeout)
{
  int ret;

  /* tcdrain is a cancellation point */

  if (cancelable && enter_cancellation_point())
    {
#ifdef CONFIG_CANCELLATION_POINTS
      /* If there is a pending cancellation, then do not perform
       * the wait.  Exit now with ECANCELED.
       */

      leave_cancellation_point();
      return -ECANCELED;
#endif
    }
```
also has it

Reply:
Done